tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/IdResolutionSpec.scala

```diff
@@ -0,0 +1,121 @@
+      _ <- aclDsl.addPermission("/", Bob, Organizations.Create)
+      _ <- adminDsl.createOrganization(org1, org1, Bob)
+      _ <- adminDsl.createProject(org1, proj11, kgDsl.projectJson(name = proj11), Bob)
+      _ <- adminDsl.createProject(org1, proj12, kgDsl.projectJson(name = proj12), Bob)
+    } yield ()
+
+    val createResources = for {
+      _ <- deltaClient.post[Json](s"/resources/$ref11/_/", resourcePayload1, Bob)(expectCreated)
+      _ <- deltaClient.post[Json](s"/resources/$ref11/_/", resourcePayload2, Bob)(expectCreated)
+      _ <- deltaClient.post[Json](s"/resources/$ref12/_/", resourcePayload2, Bob)(expectCreated)
+    } yield ()
+
+    (setup >> createResources).accepted
+  }
+
+  "Id resolution" should {
+    val iri = UrlUtils.encode("https://bluebrain.github.io/nexus/vocabulary/resource")
+
+    "lead to an authorization failure for a user without permission" in {
+      deltaClient.get[Json](s"/resolve/$iri", Alice) { (json, response) =>
+        response.status shouldEqual StatusCodes.Forbidden
+        json shouldEqual unauthorizedAccessErrorPayload
+      }
+    }
+
+    "lead to an authorization failure when trying to resolve a resource that does not exist" in {
+      deltaClient.get[Json](s"/resolve/unknownId", Bob) { (json, response) =>
+        response.status shouldEqual StatusCodes.Forbidden
+        json shouldEqual unauthorizedAccessErrorPayload
+      }
+    }
+
+    "resolve a single resource" in {
+      eventually {
+        deltaClient.get[Json](s"/resolve/${UrlUtils.encode(uniqueId)}", Bob) { (json, response) =>
+          response.status shouldEqual StatusCodes.OK
+          json shouldEqual resourcePayload1
+        }
+      }
+    }
+
+    "return search results if the same id exists across several projects" in {
+      eventually {
+        deltaClient.get[Json](s"/resolve/${UrlUtils.encode(reusedId)}", Bob) { (json, response) =>
+          response.status shouldEqual StatusCodes.OK
+          json.hcursor.get[Int]("_total") shouldEqual Right(2)
+        }
+      }
+    }
+
+    "redirect to fusion login when if text/html accept header is present (no results)" in {
+      val expectedRedirectUrl = "https://bbp.epfl.ch/nexus/web/login"
+
+      deltaClient.get[String]("/resolve/unknownId", Bob, acceptTextHtml) { (_, response) =>
+        response.status shouldEqual StatusCodes.SeeOther
+        locationHeaderOf(response) shouldEqual expectedRedirectUrl
+      }(PredefinedFromEntityUnmarshallers.stringUnmarshaller)
+    }
+
+    "redirect to fusion resource page if text/html accept header is present (single result)" in {
+      val expectedRedirectUrl =
+        s"https://bbp.epfl.ch/nexus/web/$ref11/resources/${UrlUtils.encode(uniqueId)}".replace("%3A", ":")
+
+      deltaClient.get[String](s"/resolve/${UrlUtils.encode(uniqueId)}", Bob, acceptTextHtml) { (_, response) =>
+        response.status shouldEqual StatusCodes.SeeOther
+        locationHeaderOf(response) shouldEqual expectedRedirectUrl
```

Comment:
It might be even nicer if this read:
`locationHeaderOf(response) shouldEqual fusionResourcePageFor(uniqueId)`

Reply:
Done 👍